MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -2419,77 +2568,357 @@ private static void correctNonAeroFlyerBehavior(List<Entity> entityList, int boa
+
+            // Some fighters on ground attack may be flying air cover rather than strictly
+            // air-to-ground
+            boolean isCAP = !hasGroundTargets || (Compute.d6() == 1);
+
+            // Set minimum thrust values, with lower minimums for unarmed and ground attack,
+            // and use remaining thrust to limit hardpoints
+            if (isCAP) {
+                minThrust = isUnarmed ? 2 : ((int) Math.ceil(curBomber.getWalkMP() / 2.0));
+            } else {
+                minThrust = isUnarmed ? 2 : 3;
+            }
+            maxLoad = Math.min((int) Math.floor(curBomber.getWeight() / 5.0), (curBomber.getWalkMP() - minThrust) * 5);
+
+            // Get a random percentage (40 - 90) of the maximum bomb load for armed entities
+            if (!isUnarmed) {
+                maxLoad = (int) Math.ceil((Compute.randomInt(50) + 40) * maxLoad / 100.0);
+            }
+
+            if (maxLoad == 0) {
+                continue;
+            }
+
+            // Generate bomb load
+            generatedBombs = generateExternalOrdnance(
+                    maxLoad,
+                    isCAP,
+                    isPirate,
+                    quality,
+                    campaign.getGameYear());
+            // Whoops, go yell at the ordnance technician
+            if (Arrays.stream(generatedBombs).sum() == 0) {
+                continue;
+            }
+
+            // Set a flag if any of the ordnance guided by TAG
+            hasGuided = !hasGuided &&
+                    (generatedBombs[BombType.B_LG] > 0 ||
+                            generatedBombs[BombType.B_HOMING] > 0);
+
+
+            // Store the bomb selections as we might need to add in TAG later
+            bombsByCarrier.put(i, generatedBombs);
+
+            // Do not increment bomber count for unarmed entities
+            if (!isUnarmed) {
+                numBombers++;
+            }
+
+        }
+
+        // Load bombs onto entities. If there is guided ordnance present then randomly add some TAG
+        // pods to those without the guided ordnance
+        int tagCount = Math.min(bomberList.size(), Compute.randomInt(3));
+        for (int i = 0; i < bomberList.size(); i++) {
+            Entity curBomber = bomberList.get(i);
+
+            if (bombsByCarrier.containsKey(i)) {
+                generatedBombs = bombsByCarrier.get(i);
+
+                if (hasGuided &&
+                        tagCount > 0 &&
+                        Arrays.stream(generatedBombs).sum() < (Math.floor(curBomber.getWeight() / 5)) &&
+                        generatedBombs[BombType.B_LG] == 0 &&
+                        generatedBombs[BombType.B_HOMING] == 0) {
+                    generatedBombs[BombType.B_TAG]++;
+                    tagCount--;
                 }
 
-                if (numBombers >= maxBombers) {
-                    break;
+                ((IBomber) curBomber).setBombChoices(generatedBombs);
```

Comment:
I don't like this construction:
- We're mixing bomb configuration and bomb setting; this should be separate for cleanliness and ease of parsing.
-  We duplicate this call in two places, but neither will get called if `hasGuided` is false so if there are no guided munitions, no bombs will get set on any Aero unit.
- We have a hard-coded set of guided munitions that doesn't include Aero Guided Arrow IV Bomb munitions, ; this should probably be a set or list that we declare up with the other constants.
- It looks like the second option erases all bombs for the current unit and just gives it one single TAG pod, which I guarantee is not what we want.

I would actually suggest splitting the TAG-adding code out into a separate helper that just mutates a generatedBombs array based on the current bomb load on the bomber, the bomber's weight, and perhaps the bomber's loaded speed - that way we can avoid pushing a bomber from "fast enough" to "too slow" by adding the TAG pods only to units that have a slot to spare before dropping down one thrust rating.

Reply:
The TAG part might be cleaned up a bit. The reasoning on that is, you can't use a TAG pod on the same round you're dropping bombs. So can't self-designate for laser/TAG guided bombs, and while you can self-designate for a homing Arrow that you dropped last round, the minimum movement of fixed wing aircraft means the chances they can be over the target zone the very next round is slim.  So TAG designators get pushed to any fixed wing aircraft that did NOT get any external ordnance, which happens from the random limit on how many units to load ordnance on to. That means units with external TAG will only have the one unit and will always have sufficient thrust; it also means that if all units are loaded with ordnance any guided ordnance will need ground designators or built-in TAG.